src/app.jsx

```diff
@@ -91,18 +99,25 @@ class Application extends React.Component {
                     this._asyncRequestContainers = null;
                     this.setState({containersMeta: reply.containers || []});
                     this.state.containersMeta.map((container) => {
-                        utils.varlinkCall(utils.PODMAN, "io.podman.InspectContainer", JSON.parse('{"name":"' + container.id + '"}'))
+                        utils.varlinkCall(utils.PODMAN, "io.podman.InspectContainer", {name: container.id})
+
                                 .then(reply => {
-                                    const temp_containers = this.state.containers;
+                                    let temp_containers = this.state.containers;
                                     temp_containers.push(JSON.parse(reply.container));
```

Comment:
I didn't spot that in the previous review: This modifies `this.state` directly, which is invalid. This needs to make a copy, with e. g. `temp_containers = this.state.containers.slice()`. Also, I don't get the simple `push()` - when you call it multiple times, you would never remove the old containers from the state; might be better to make the state an id → props mapping instead of a simple list?

Reply:
I also tried to make it a map at first ```temp_containers[container.id] = JSON.parse(reply.container);``` but in JS the array length is 0 because the nonsequence index.  not rendered. I think the reason is in ``` cockpit-components-listing.jsx``` Ln299, it checked ```if (!props.children || props.children.length === 0)```. So I just use array and use ```findIndex()``` if needed. Do you have any ideas about this?

If keep it as an array, can I use this instead of slice()?
```
this.setState(preState => ({
    containers: [...preState.containers, JSON.parse(reply.container)]
}));
```

Reply:
If you turn this into a map, then of course you have to change the render() function of the component that receives this as property (Containers.jsx?) accordingly, to iterate over the dictionary instead of the map. If you need the length, you can use `Object.keys(state.containers).length`.

You can use the spread operator instead of `.slice()` to copy an array, sure. But as I said, merely appending this is not enough - you will have to remove the old entries for the same ID first.